src/gt4py/eve/datamodels/core.py

```diff
@@ -1291,7 +1291,7 @@ def _make_datamodel(  # noqa: C901  # too complex but still readable and documen
 
 @utils.optional_lru_cache(maxsize=None, typed=True)
 def _make_concrete_with_cache(
-    datamodel_cls: Type[GenericDataModelT],
+    datamodel_cls: Type,
```

Comment:
what's the problem with the concrete type?

Reply:
I must admit that I do not fully understand but it solves this error:
`src/gt4py/eve/datamodels/core.py:817:9: error: Argument 1 to "_make_concrete_with_cache" has incompatible type "type[GenericDataModelT]"; expected "type[Never]"  [arg-type]`